src/event_parser.py

```diff
@@ -0,0 +1,86 @@
+from datetime import datetime
+from event_group_rule import EventGroupRule
+from log_event_group import LogEventGroup
+from log_parser import LogEvent, LogParser
+
+""" Module for parsing Log Events into Event Groups """
+
+class LogEventParser:
+    """ Parser for applying group rules against a list of events to
+        create instances of event groups
+        
+        Attributes
+        ----------
+        log_events: list of LogEvent objects to be parsed
+        event_group_rules: list of EventGroupRule objects to be applied
+                           for event grouping
+        event_groups_found: chronological list list of detected 
+                            LogEventGroup objects
+        event_groups_found_dict: dictionary of LogEventGroups where
+                                 key is the group tag """
+
+    def __init__(self, log_events, event_group_rules):
+        self.log_events = log_events
+        self.event_group_rules = event_group_rules
+
+        self.event_groups_found = []
+        self.event_groups_found_dict = {}
+
+    def get_event_search_dict(self, event_group_rule):
```

Comment:
I think all these functions could use some docstrings. Some of them seem pretty complicated.

Reply:
Oof. Yes, agreed, this is all pretty dense.

Reply:
Docstring comments should go after the definition, not before it. It works similarly to the way class docstring work.